### PR TITLE
Update NotThrowAfter tests to use FakeClock (GH-1116) 

### LINKED ISSUE #1116: NotThrowAfter tests does not use FakeClock
### Description

Two tests of `NotThrowAfter` and `NotThrowAfterAsync`, respectively, does not utilize the new `FakeClock` introduced in #1076 and are sensitive to timing.

* `When_no_exception_should_be_thrown_for_async_func_executed_with_wait_after_wait_time_but_it_was_it_should_throw`
* `When_no_exception_should_be_thrown_for_async_func_executed_with_wait_after_wait_time_and_none_was_it_should_not_throw`
* `When_no_exception_should_be_thrown_for_async_func_after_wait_time_but_it_was_it_should_throw`
* `When_no_exception_should_be_thrown_for_async_func_after_wait_time_and_none_was_it_should_not_throw`

Let's fix 'em!

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #1116

### CODE REVIEW

File: Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs [hunk 1]
```diff
@@ -1177,13 +1183,17 @@ public void When_no_exception_should_be_thrown_for_async_func_after_wait_time_bu
...
             // Arrange
             //-----------------------------------------------------------------------------------------------------------
-            var watch = Stopwatch.StartNew();
             var waitTime = 2.Seconds();
             var pollInterval = 10.Milliseconds();
 
+            var clock = new FakeClock();
+            var timer = clock.StartTimer();
+            clock.Delay(waitTime);
+            clock.CompletesBeforeTimeout();
```
Comment (Evangelink) [inline Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs]: This test seems to be the only one requiring this call (note that the RunsIntoTimeout method) also works here.

File: Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs [hunk 2]
```diff
@@ -1033,13 +1036,13 @@ public void
...
             //-----------------------------------------------------------------------------------------------------------
-            Action action = () => throwLongerThanWaitTime.Should()
+            Action action = () => throwLongerThanWaitTime.Should(clock)
                 .NotThrowAfter(waitTime, pollInterval, "we passed valid arguments");
 
             //-----------------------------------------------------------------------------------------------------------
             // Assert
             //-----------------------------------------------------------------------------------------------------------
-            action.Should().Throw<XunitException>()
+            action.Should(clock).Throw<XunitException>()
```
Comment (jnyrup) [inline Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs]: 🤔 `Should().Throw()` and `Should().NotThrow()` should not need an `ITimer` as they aren't timing related assertions?

File: Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs (hunk 1 above)
Comment (jnyrup) [reply Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs]: Whether you use `CompletesBeforeTimeout` or `RunsIntoTimeout` seems to only matter for sync assertions, e.g. `CompleteWithin()`.
See `When_task_completes_fast_it_should_succeed`.

I frankly don't known if that is intended.

File: Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs (hunk 2 above)
Comment (Evangelink) [reply Tests/Shared.Specs/AsyncFunctionExceptionAssertionSpecs.cs]: Indeed!

**Pull request merged**